controllers/azuremanagedcontrolplane_reconciler.go

```diff
@@ -78,8 +82,8 @@ func newAzureManagedControlPlaneReconciler(scope *scope.ManagedControlPlaneScope
 			subnetsSvc,
 			managedClustersSvc,
 			privateEndpointsSvc,
-			tags.New(scope),
 			resourceHealthSvc,
+			tagsSvc,
 		},
```

Comment:
Was flipping the order here intentional? In this order, if a cluster gets stuck in an unhealthy state as flagged by resource health, then its tags won't update until it becomes healthy again.

Reply:
No, it was probably a mistake in rebasing. I'll restore the order.